### DISCUSSION TITLE
compilation issue in Windows(Visual studio) due to memcpy_s

### DISCUSSION THREAD

**Discussion by ghost** [open-source-parsers/jsoncpp#1331] [Q&A]
Hi, 
memset_s function is introduced in below commit.
https://github.com/open-source-parsers/jsoncpp/commit/30170d651c108400b1b9ed626ba715a5d95c5fd2

But memset_s is not available in windows API. This causes compilation in Visual studio to break.

The previous implementation to securely zero out memory(Volatile pointer with memset) will do the same as memset_s function. ie.) Compiler won't optimize operations made on volatile data. 

Can we revert back to memset (with volatile pointer) ?

**Comment by BillyDonahue** [comment]
I have no really opinion on `memset` or `memset_s` right now, but the `volatile` was worthless and should not be restored.
See PR #1228.

"I think we want the C11 `memset_s` on platforms on which it's available."

It sounds like we just need better detection logic for determining those platforms.
Maybe Meson knows how to check for it autoconf style. I don't know Meson though.
